### ISSUE TITLE
Custom Dataset

### ISSUE THREAD

**Issue by Zumbalamambo** [eric-yyjau/pytorch-superpoint#9]
How do I prepare the custom dataset for training this model?

**Comment by eric-yyjau** [comment]
Hi @Zumbalamambo ,

Good question. Basically you need to do step 2), 3) to train your model.
Step 2) help you export the pseudo ground truth.

Good new is that both steps use the same data loader.
I'll suggest you to modify from `Kitti_inh.py` (https://github.com/eric-yyjau/pytorch-superpoint/blob/master/datasets/Kitti_inh.py), which is inherited from `Coco.py`.

You need to prepare several things: (let's use kitti dataset as an example)
1. the dataset it self
2. the train/val split txt files

Then start to edit the dataloader:
https://github.com/eric-yyjau/pytorch-superpoint/blob/86b9621888e7c91801a5e9de889009573f407e45/datasets/Kitti_inh.py#L99
Edit to fit your dataset, so that you can get image paths (https://github.com/eric-yyjau/pytorch-superpoint/blob/86b9621888e7c91801a5e9de889009573f407e45/datasets/Kitti_inh.py#L119), scene_name and name (https://github.com/eric-yyjau/pytorch-superpoint/blob/86b9621888e7c91801a5e9de889009573f407e45/datasets/Kitti_inh.py#L191) can help you save the exported labels.

KITTI dataloader is over-complicated as it is used for other applications as well. You can ignore things unrelated to images and folders.

Hope this helps.
